### PR TITLE
Add CJS support to connect-extension-protocol

### PR DESCRIPTION
**Pull request by wirednkod**
This PR adds CJS support to `connect-extension-protocol`.
It is needed for adding CJS support to `@substrate/connect` 

This needs to merge prior to #1142 

This also needs to be merged AFTER [#2487](https://github.com/paritytech/smoldot/pull/2487)

### CODE REVIEW

File: packages/connect-extension-protocol/fix-package-type [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+{
+    "type": "commonjs"
+}
+!EOF
+
+cat >dist/mjs/package.json <<!EOF
+{
+    "type": "module"
+}
+!EOF
```
Comment (tomaka) [inline packages/connect-extension-protocol/fix-package-type:16]: Why is this needed? There's already a `package.json`, why do we need two additional ones?

Comment (wirednkod) [reply packages/connect-extension-protocol/fix-package-type:16]: Its not exactly "two additional ones", rather "two  complementary ones". These `package.json` files define the default package type for each of the `dist/*` sub-directories. Main `package.json` does not have a type property (if it did - hybrid solution would not work), and with this script it is pushed down to the `package.json` files under `dist/`.

Comment (tomaka) [reply packages/connect-extension-protocol/fix-package-type:16]: It doesn't answer my question: _why_ do we need them?

Comment (wirednkod) [reply packages/connect-extension-protocol/fix-package-type:16]: When `type` is set to `module` it clarifies that this is an ESM package - we could avoid that by changing the `js` extension to `mjs` but I find it unnecessary as a simple `"type" : "module"` would do the same.

File: packages/connect-extension-protocol/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
 # Changelog
 
 ## [Unreleased]
+
+### Added
+
+- Add CJS support to @substrate/connect-extension-protocol ([#1143](https://github.com/paritytech/substrate-connect/pull/1143))
```
Comment (tomaka) [inline packages/connect-extension-protocol/CHANGELOG.md:7]:
```suggestion
- Add CommonJS support to @substrate/connect-extension-protocol ([#1143](https://github.com/paritytech/substrate-connect/pull/1143))

```

File: packages/connect/tsconfig.json [hunk 3]
```diff
@@ -22,5 +22,5 @@
     "src/connector/specs/*.json"
```
Comment (tomaka) [inline packages/connect/tsconfig.json:22]: Can you remove `outDir`, `module` and `target` from `tsconfig.json`?

File: packages/connect/tsconfig.json [hunk 4]
```diff
@@ -22,5 +22,5 @@
...
   ],
   "include": ["src/**/*", "src/**/*.json"],
-  "references": [{ "path": "../connect-extension-protocol" }]
+  "references": [{ "path": "../connect-extension-protocol/tsconfig-mjs.json" }]
```
Comment (tomaka) [inline packages/connect/tsconfig.json:25]: Why is that needed?

File: packages/connect-extension-protocol/tsconfig-mjs.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "esnext",
+    "outDir": "./dist/mjs",
+    "target": "esnext"
```
Comment (tomaka) [inline packages/connect-extension-protocol/tsconfig-mjs.json:6]:
```suggestion
    "target": "es2018"
```

Let's keep the values that were in `tsconfig.json`

File: packages/connect-extension-protocol/tsconfig-mjs.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "esnext",
```
Comment (tomaka) [inline packages/connect-extension-protocol/tsconfig-mjs.json:4]:
```suggestion
    "module": "es2020",
```

File: packages/connect-extension-protocol/fix-package-type [hunk 7]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (tomaka) [inline packages/connect-extension-protocol/fix-package-type:1]: Can you add a `.sh` extension to this file name?

**Review by tomaka — CHANGES_REQUESTED** [review]

**Review by tomaka — APPROVED** [review]
